network_wrangler/roadway/nodes/create.py

```diff
@@ -145,6 +145,7 @@ def _create_nodes_from_link(
     nodes_df["geometry"] = nodes_df["geometry"].apply(
         get_point_geometry_from_linestring, pos=link_pos
     )
+    nodes_df = gpd.GeoDataFrame(nodes_df, geometry='geometry')
```

Comment:
nodes_df is not a geodataframe?

Reply:
Its possible that its not if the geometry column is malformed or if it is coming from a function that is creating a new node. - so this makes sure it is.